js/ui/animation.js

```
@@ -75,6 +84,23 @@ var Animation = new Lang.Class({
 
     _onDestroy: function() {
         this.stop();
+    },
+
+    setFrameTimeouts: function(timeoutList) {
+        if (timeoutList.length == 0) {
+            log('Cannot set an empty list as the frame-timeouts for animation!');
```

This should throw an error.